sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1580,6 +1647,59 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
     }
   }
 
+  private[this] def castDayTimeIntervalToIntegralTypeCode(
+      ctx: CodegenContext,
+      endFiled: Byte,
+      integralType: String,
+      catalogType: String): CastFunction = {
+    val util = IntervalUtils.getClass.getCanonicalName.stripSuffix("$")
+    integralType match {
```

Comment:
I think the caller side should just pass `Int`, `Long`, etc. and here we can simply write `$util.dayTimeInterval$integralType`